tests/benchmarks/peripheral_stress_test/src/gpio_thread.c

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(gpio, LOG_LEVEL_INF);
+
+#include <zephyr/drivers/gpio.h>
+#include "common.h"
+
+#define GPIO_NODE		DT_ALIAS(led2)
+#if !DT_NODE_HAS_STATUS(GPIO_NODE, okay)
+#error "Unsupported board: led2 devicetree alias is not defined"
+#endif
+
+static const struct gpio_dt_spec gpio_spec = GPIO_DT_SPEC_GET(GPIO_NODE, gpios);
+
+/* GPIO thread */
+static void gpio_thread(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	int ret;
+
+	atomic_inc(&started_threads);
+
+	if (!gpio_is_ready_dt(&gpio_spec)) {
+		LOG_ERR("Device %s is not ready.", gpio_spec.port->name);
+		atomic_inc(&completed_threads);
```

Comment:
Not needed here? missing increment will allow to detect that something has failed

Reply:
Failure will be detected based on missing log LOG_INF("GPIO thread has completed");